### ISSUE TITLE
How can i recognition space in OCR model ?

### ISSUE THREAD

**Issue by ThorPham** [open-mmlab/mmocr#580]
I add space character in classes OCR ( in DICT90) and training custom dataset. But when i inference space is not recognition .
Please let me know some how can i config model to recognition space character . Thank you

**Comment by gaotongxiao** [comment]
MMOCR skips space character even if you have added it into the dictionary. You can replace the space with a special token (which is not included by DICT90 before) for training and convert it back during inference.

**Comment by ThorPham** [comment]
@gaotongxiao Thank you so much. I have other question. I want to apply some my augmentation for OCR model. Where  i can modify code for apply new augment method .

**Comment by gaotongxiao** [comment]
Augmentations are one kind of transform in MMOCR. You can add your augmentation class to mmocr/datasets/pipelines/ocr_transforms.py. Make sure you have done the following things:

1. Use the `@PIPELINES.register_module()` wrapper for your transform;
2. Implement your augmentation operations in `def __call__(self, results)`. The image ndarray should be inside `results['img']`. If your augmentation involves shape change, don't forget to update `results['img_shape']`. You can find many reference implementations like `ResizeOCR` in the same file.
3. Add your transform to `train_pipeline` in the config. Take NRTR as an example, it should be likely to be right after `ResizeOCR`: https://github.com/open-mmlab/mmocr/blob/3707d671061f4d75ba4d7357e8908fabb59f9798/configs/textrecog/nrtr/nrtr_r31_1by8_1by4_academic.py#L30-L47. It's because augmentations are generally applied to the picture before tentorized and normalized.

We know that input and output of each transform are unclear for now and it's hard to follow as a beginner, so we are working hard to supplement docstrings and writing a general introduction to transforms. For now, you might refer to [MMDet's intro to transforms] to gain a better understanding of it.

**Comment by ThorPham** [comment]
@gaotongxiao thank a lot. I will try .
I want to change threshold for inference   DBnet for text detection model. How can i set this .

**Comment by EMRAN-SALEH-CORSEARCH** [comment]
Hi, 
Can I know where in the code the MMOCR skips the space character? 
Thank you!

**Comment by pbamotra** [comment]
https://mmocr.readthedocs.io/en/latest/tutorials/blank_recog.html
